src/input.rs

```diff
@@ -130,9 +132,21 @@ impl<'a> Input<'a> {
 
     fn _ordinary_file(path: &Path) -> Self {
         let kind = InputKind::OrdinaryFile(path.to_path_buf());
+        let metadata = match fs::metadata(path.to_path_buf()) {
+            Ok(meta) => {
+                let size = meta.len();
+
+                InputMetadata {
+                    size: Some(size),
+                    ..InputMetadata::default()
+                }
+            }
+            Err(_) => InputMetadata::default(),
+        };
+
```

Comment:
Hmm this comment got lost so I'm making it again. You can simplify this code like this:
```suggestion
        let metadata = InputMetadata {
            size: fs::metadata(path).map(|m| m.len()).ok(),
            ..InputMetadata::default()
        };

```